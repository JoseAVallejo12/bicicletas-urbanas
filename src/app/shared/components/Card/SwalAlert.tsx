import Swal from 'sweetalert2';

const alquilerNoRegistrado = (mensaje: string) => {
  Swal.fire({
    icon: 'error',
    title: 'Oops...',
    text: mensaje,
  });
}

const alquilerResgistrado = () => {
  Swal.fire({
    position: 'top',
    icon: 'success',
    title: 'Your work has been saved',
    showConfirmButton: false,
    timer: 1500
  })
}

export default {
  alquilerNoRegistrado,
  alquilerResgistrado
}