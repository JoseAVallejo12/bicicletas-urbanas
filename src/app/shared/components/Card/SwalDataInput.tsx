import Swal, { SweetAlertResult } from 'sweetalert2';

export const obtenerCedulaUsuario = (): Promise<SweetAlertResult<any>> => {
  return (Swal.fire({
    title: 'escriba cc',
    input: 'number'
  }))
}