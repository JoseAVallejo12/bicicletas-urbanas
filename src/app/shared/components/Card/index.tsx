import React, { useEffect, useState } from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import { servcioCrearAlquiler } from '../../../core/api/alquiler.repositorio';
import { obtenerCedulaUsuario } from './SwalDataInput';
import Alertas from './SwalAlert';
import { Bicicleta } from 'app/feature/Bicicleta/models/Bicicleta';


const useStyles = makeStyles({
  root: {
    maxWidth: 345,
  },
  media: {
    height: 300,
  },
});

interface BicicletaProps {
  bici: Bicicleta;
  listarBici: () => void;
}

export const MediaCardBici: React.FC<BicicletaProps> = ({bici, listarBici}) => {
  const [alquilerControl, setBotonAlquiler] =  useState(false);
  const classes = useStyles();
  const imagenPath = require(`../../../../assets/img/bicicletas/${bici.id}.jpg`)

  const handleClickAlquiler = async () => {
    const res = await obtenerCedulaUsuario();
    if (res.isConfirmed && res.value > 0) {
      const respuesta = await servcioCrearAlquiler(res.value, bici.id.toString())
      if (respuesta.status !== 201) {
        respuesta.json().then(({message}) => Alertas.alquilerNoRegistrado(message))
      } else {
        listarBici();
        setBotonAlquiler(true);
        Alertas.alquilerResgistrado();
      }
    }
  };

  useEffect(()=>{
    if (bici.estado !== 'libre') {
      setBotonAlquiler(true);
    }
  }, [])

  return (
    <Card className={classes.root}>
      <CardActionArea>
        <CardMedia
          className={classes.media}
          image={imagenPath}
        />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h4">
            <b>Marca:</b> {bici.marca}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            Estado: {bici.estado} <br/>
            Precio: $ {bici.valorHora} COP c/hr <br/>
            Descripcion: {bici.descripcion}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary" onClick={handleClickAlquiler} disabled={alquilerControl} >
          Alquilar
        </Button>
        <Button size="small" color="primary" disabled={alquilerControl}>
          Reservar
        </Button>
      </CardActions>
    </Card>
  );
}
