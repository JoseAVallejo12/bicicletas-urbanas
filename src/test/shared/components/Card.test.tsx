import * as React from 'react';
import { ShallowWrapper, shallow } from 'enzyme';

import { MediaCardBici } from '../../../app/shared/components/Card/index';

describe(' <MediaCardBici />Test', () => {
  let componentWrapper: ShallowWrapper;
  const bicicleta = {
    id: 1,
    marca: 'Jeep',
    serial: 'BALTORO 29 V20',
    color: 'Roja',
    almacenActual: 1,
    fechaCompra: '2021-01-19T05:00:00.000Z',
    estado: 'alquilada',
    valorHora: '5000',
    descripcion: 'Bicicleta de Montaña Jeep Baltoro 29 pulgadas Género: Unisex Material del marco: Aluminio, Número de cambios: 21, Tipo de freno: Disco mecánico, Suspensión: Delantera'
  };

  beforeEach(() => {
    componentWrapper = shallow(<MediaCardBici bici={bicicleta} listarBici={() => {}}/>)
  })

  afterEach(() => {
    componentWrapper.unmount();
  });

  it('should match snapshot', () => {
    expect(componentWrapper).toMatchSnapshot();
  });

  it('renders the heading', () => {
    expect(componentWrapper.find('.MuiPaper-root')).toHaveLength(0);
  });


});

export {};
