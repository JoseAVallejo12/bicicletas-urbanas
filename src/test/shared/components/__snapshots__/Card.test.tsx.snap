// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` <MediaCardBici />Test should match snapshot 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardActionArea))>
    <WithStyles(ForwardRef(CardMedia))
      className="makeStyles-media-2"
      image="1.jpg"
    />
    <WithStyles(ForwardRef(CardContent))>
      <WithStyles(ForwardRef(Typography))
        component="h4"
        gutterBottom={true}
        variant="h5"
      >
        <b>
          Marca:
        </b>
         
        Jeep
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        color="textSecondary"
        component="p"
        variant="body2"
      >
        Estado: 
        alquilada
         
        <br />
        Precio: $ 
        5000
         COP c/hr 
        <br />
        Descripcion: 
        Bicicleta de Montaña Jeep Baltoro 29 pulgadas Género: Unisex Material del marco: Aluminio, Número de cambios: 21, Tipo de freno: Disco mecánico, Suspensión: Delantera
      </WithStyles(ForwardRef(Typography))>
    </WithStyles(ForwardRef(CardContent))>
  </WithStyles(ForwardRef(CardActionArea))>
  <WithStyles(ForwardRef(CardActions))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      onClick={[Function]}
      size="small"
    >
      Alquilar
    </WithStyles(ForwardRef(Button))>
    <WithStyles(ForwardRef(Button))
      color="primary"
      disabled={false}
      size="small"
    >
      Reservar
    </WithStyles(ForwardRef(Button))>
  </WithStyles(ForwardRef(CardActions))>
</WithStyles(ForwardRef(Card))>
`;
